integrations/mcp/src/haystack_integrations/tools/mcp/mcp_toolset.py

```diff
@@ -0,0 +1,232 @@
+import logging
+from typing import Any
+
+import httpx
+from exceptiongroup import ExceptionGroup
+from haystack.core.serialization import generate_qualified_class_name, import_class_by_name
+from haystack.tools import Tool, Toolset
+
+from .mcp_tool import (
+    AsyncExecutor,
+    MCPConnectionError,
+    MCPServerInfo,
+    MCPToolNotFoundError,
+    SSEServerInfo,
+    StdioServerInfo,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class MCPToolset(Toolset):
+    """
+    A Toolset that connects to an MCP (Model Context Protocol) server and provides
+    access to its tools.
+
+    MCPToolset dynamically discovers and loads all tools from any MCP-compliant server,
+    supporting both network-based SSE connections and local process-based stdio connections.
+    This dual connectivity allows for integrating with both remote and local MCP servers.
+
+    Example using MCPToolset in a Haystack Pipeline:
```

Comment:
Could we add to this example all steps needed to get it to run? I just tried copy-pasting and ran into an issue about `uvx` not being found. 

It'd be great if we could suggest what keys and additional things you need to install to be able to use this example as comments within the python code block.